great_expectations/core/batch.py

```diff
@@ -914,7 +914,7 @@ def _is_fluent_batch_request(args: dict[str, Any]) -> bool:
 
 
 def batch_request_contains_batch_data(
-    batch_request: Optional[Union[BatchRequestBase, dict]] = None
+    batch_request: Optional[Union[BatchRequestBase, FluentBatchRequest, dict]] = None
```

Comment:
@Kilo59 I am not sure if this is needed; do we already have `FluentBatchRequest` ability to accept `batch_data`?  I do recall a PR being in the works to allow `batch_request.dataframe` to be set, but was not sure.  Thanks.

Reply:
I'm not sure about the other points you raise but this is needed.